File: services/business-api/src/main/java/com/smartfarm/inventory/masterdata/domain/MasterDataChanges.java
package com.smartfarm.inventory.masterdata.domain;

import java.util.List;
import java.util.UUID;

public record MasterDataChanges(
        String cursor,
        boolean fullResyncRequired,
        List<NamedEntity> organizations,
        List<NamedEntity> buildings,
        List<NamedEntity> pens,
        List<DeletedEntity> deletedEntities) {
    public record NamedEntity(
            UUID id,
            UUID parentId,
            String code,
            String name,
            boolean enabled,
            String erpExternalId,
            long syncVersion) {
    }

    public record DeletedEntity(String entityType, UUID id, long syncVersion) {
    }
}

File: apps/mobile/lib/features/master_data/domain/master_data_changes.dart
class MasterDataChanges {
  const MasterDataChanges({
    required this.cursor,
    required this.fullResyncRequired,
    required this.organizations,
    required this.buildings,
    required this.pens,
    required this.deletedEntities,
  });

  final String cursor;
  final bool fullResyncRequired;
  final List<MasterDataEntity> organizations;
  final List<MasterDataEntity> buildings;
  final List<MasterDataEntity> pens;
  final List<DeletedMasterDataEntity> deletedEntities;
}

class MasterDataEntity {
  const MasterDataEntity({
    required this.id,
    required this.parentId,
    required this.code,
    required this.name,
    required this.enabled,
    required this.syncVersion,
  });

  final String id;
  final String? parentId;
  final String code;
  final String name;
  final bool enabled;
  final int syncVersion;
}

class DeletedMasterDataEntity {
  const DeletedMasterDataEntity({required this.entityType, required this.id});
  final String entityType;
  final String id;
}

class PenChoice {
  const PenChoice({
    required this.organizationId,
    required this.buildingId,
    required this.buildingCode,
    required this.buildingName,
    required this.penId,
    required this.penCode,
    required this.penName,
    required this.enabled,
  });

  final String organizationId;
  final String buildingId;
  final String buildingCode;
  final String buildingName;
  final String penId;
  final String penCode;
  final String penName;
  final bool enabled;
}
